Sources/Brave/Frontend/Player/PlayerUtils.swift

```diff
@@ -0,0 +1,64 @@
+// Copyright 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import BraveShared
+import Shared
+import Strings
+import BraveShields
+
+class PlayerUtils {
+  private static let youtubePrefix = "youtube"
+  
+  static func makeBravePlayerURL(from url: URL) -> URL? {
+    guard let videoInfo = prefixAndVideoID(from: url) else { return nil }
+    guard let encodedID = videoInfo.videoID.addingPercentEncoding(withAllowedCharacters: .urlPathAllowed) else { return nil }
```

Comment:
`(withAllowedCharacters: .urlHostAllowed` instead of `withAllowedCharacters: .urlPathAllowed`. Because URL path symbols is what we want to encode in the first place.

Reply:
Ok done